orchagent/orch.cpp

```diff
@@ -373,10 +356,17 @@ void Orch::addConsumer(DBConnector *db, string tableName)
 {
     if (db->getDB() == CONFIG_DB)
     {
-        Consumer consumer(new SubscriberStateTable(db, tableName));
-        m_consumerMap.insert(ConsumerMapPair(tableName, consumer));
-    } else {
-        Consumer consumer(new ConsumerStateTable(db, tableName, gBatchSize));
-        m_consumerMap.insert(ConsumerMapPair(tableName, consumer));
+        addExecutor(tableName, new Consumer(new SubscriberStateTable(db, tableName), this));
```

Comment:
Just to understand, since there is no return pointer to object here, the free must be happening in the callee, right?. Also will it be good to use smart pointers (unique_ptr) to avoid any memory leaks?

Reply:
The Executor classes (include Consumer) will hold and responsible to release the memory by design. You may check the comment there.

The Orch classes will hold and responsible to release the memory by original design.

I would not use unique_ptr here, since the lifecycle of the new objects is larger than the function-local.

---
In reply to: [150727650](https://github.com/Azure/sonic-swss/pull/382#discussion_r150727650) [](ancestors = 150727650)

Reply:
ok